sycl/include/sycl/detail/pi.h

```diff
@@ -2027,6 +2027,16 @@ __SYCL_EXPORT pi_result piTearDown(void *PluginParameter);
 /// timestamp
 __SYCL_EXPORT pi_result piPluginGetLastError(char **message);
 
+/// API to get backend specific optimization option.
+/// \param opt_level is an integer that contains frontend optimization level.
+/// \param backend_option is used to return the backend optimization option
+/// corresponding to frontend optimization level.
+///
+/// \return PI_SUCCESS is returned always. If a valid option is not
+/// available, an empty string is returned.
+__SYCL_EXPORT pi_result
+piPluginGetBackendOptimizationOption(int opt_level, char **backend_option);
```

Comment:
Should we generalize this to translate from any front-end option to backend option? E.g. translate "-Od" (and -O0") to "-ze-opt-disable" (for L0 backend)

Reply:
Currently, we aim to support only O0 to O3. Having said that, this implementation relies on the optimization level returned by the front-end driver and the front-end driver should be able to emit the correct optimization level for a given front-end option.

Thanks

Reply:
> Currently, we aim to support only O0 to O3.

That's fine, you can only support "O0" to "O3" only initially.
What I suggest is that the interface is generalized to be string -> string, not int -> string

Note, there are other options that are currently hardcoded but should better be moved to this interface:
https://github.com/intel/llvm/pull/8763/files#diff-78dd7f7ba0b6120dece1ae4ab5a09c9936ff654a1de2c31ff2dbb1fc58d90393R410

Reply:
That's a good point. I would prefer to add that as part of a separate PR as this one is bloating up a bit. Hope that's ok.

Thanks

Reply:
I'd suggest doing it in this original change-set especially if you are going to make it platform-based anyway. That should not be a big change.